ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/DownloadHeadersStep.java

```diff
@@ -58,13 +64,20 @@ public DownloadHeadersStep(
 
   private CompletableFuture<List<BlockHeader>> downloadHeaders(
       final CheckpointRange checkpointRange) {
+    LOG.debug(
+        "Downloading headers for range {} to {}",
+        checkpointRange.getStart().getNumber(),
+        checkpointRange.getEnd().getNumber());
+    if (checkpointRange.getSegmentLengthExclusive() == 0) {
```

Comment:
(optional) We could also update `DownloadHeaderSequenceTask` to just return the reference header if the segment length is 0.

Reply:
We could but it would wind up being a bit late in the process - after the peer has been selected etc.